avaje-config/src/main/java/io/avaje/config/InitialLoader.java

```diff
@@ -253,8 +257,17 @@ private void loadViaSystemProperty() {
   }
 
   boolean loadWithExtensionCheck(String fileName) {
+
+    // no need to custom URL load regular cp and file schemes
+    fileName = fileName.replace("classpath:/", "").replace("file:/", "");
+    if (fileName.contains(":/")) {
```

Comment:
This method and what it calls looks to be the most dangerous so I think we should have some unit tests that exercise this bit.

Reply:
I realize you are probably adding the unit tests later but just as a reminder to really make sure there is lots of tests for this part.

Reply:
naturally

Reply:
This part is concerning:

```java
    fileName = fileName.replace("classpath:/", "").replace("file:/", "");
```

Ideally you would do `fileName.replaceStart` but I can't remember if that is 11 or not so you may have to write your own replaceStart.